slobrok/src/vespa/slobrok/server/rpc_server_manager.cpp

```diff
@@ -199,84 +191,80 @@ RpcServerManager::remove(ManagedRpcServer *rpcsrv)
 
 
 OkState
-RpcServerManager::removeRemote(const char *name,
-                               const char *spec)
+RpcServerManager::removeRemote(const std::string &name, const std::string &spec)
 {
-    NamedService *old = _rpcsrvmap.lookup(name);
-    if (old == NULL) {
+    const NamedService *old = _rpcsrvmap.lookup(name);
+    if (old == nullptr) {
         // was alright already, remove any reservation too
         _rpcsrvmap.removeReservation(name);
         return OkState(0, "already done");
     }
-    if (strcmp(old->getSpec(), spec) != 0) {
+    if (old->getSpec() != spec) {
         return OkState(1, "name registered, but with different spec");
     }
-    NamedService *td = _rpcsrvmap.remove(name);
-    LOG_ASSERT(td == old);
-    delete td;
+    std::unique_ptr<NamedService> td = _rpcsrvmap.remove(name);
+    LOG_ASSERT(td.get() == old);
     return OkState(0, "done");
 }
 
 OkState
-RpcServerManager::removeLocal(const char *name, const char *spec)
+RpcServerManager::removeLocal(const std::string & name, const std::string &spec)
 {
-    NamedService *td = _rpcsrvmap.lookup(name);
-    if (td == NULL) {
+    const NamedService *td = _rpcsrvmap.lookup(name);
+    if (td == nullptr) {
         // already removed, nop
         return OkState();
     }
 
-    RemoteSlobrok *partner = _exchanger.lookupPartner(name);
-    if (partner != NULL) {
+    const RemoteSlobrok *partner = _exchanger.lookupPartner(name);
+    if (partner != nullptr) {
         return OkState(13, "cannot unregister partner slobrok");
     }
 
-    ManagedRpcServer *rpcsrv = _rpcsrvmap.lookupManaged(name);
-    if (rpcsrv == NULL) {
+    const ManagedRpcServer *rpcsrv = _rpcsrvmap.lookupManaged(name);
+    if (rpcsrv == nullptr) {
         return OkState(13, "not a local rpcserver");
     }
 
-    if (strcmp(rpcsrv->getSpec(), spec) != 0) {
+    if (rpcsrv->getSpec() != spec) {
         // the client can probably ignore this "error"
         // or log it on level INFO?
         return OkState(1, "name registered, but with different spec");
     }
-    td = _rpcsrvmap.remove(name);
-    LOG_ASSERT(td == rpcsrv);
-    delete rpcsrv;
+    auto tdUP = _rpcsrvmap.remove(name);
+    LOG_ASSERT(tdUP.get() == rpcsrv);
     _exchanger.forwardRemove(name, spec);
     return OkState();
 }
 
 
 void
-RpcServerManager::addManaged(const char *name, const char *spec,
-                             RegRpcSrvCommand rdc)
+RpcServerManager::addManaged(const std::string &name, const std::string &spec, RegRpcSrvCommand rdc)
```

Comment:
Command -> Data

Reply:
Not possible as Data is visible outside of cmd.cpp